services/merak-topo/handler/handler.go

```diff
@@ -429,39 +436,44 @@ func Info(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.Retu
 			}
 
 			returnMessage.ComputeNodes = append(returnMessage.ComputeNodes, &crm)
-
 		}
-
 	}
 
-	go UpdateComputenodeInfo(k8client, topo_id)
+	go UpdateComputenodeInfo(k8client, topoPrefix, namespace)
 
-	return nil
+	if err_flag == 1 {
+		err_return = errs
+	} else {
+		err_return = nil
+	}
+	return err_return
 }
 
-func Delete(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.ReturnTopologyMessage) error {
-	topo, err_db := database.FindTopoEntity(topo_id, "")
+func Delete(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.ReturnTopologyMessage, topoPrefix string, namespace string) error {
+
+	topo, err_db := database.FindTopoEntity(topoPrefix, "")
 
 	if err_db != nil {
-		return fmt.Errorf("query topology_id error %s", err_db)
+		utils.Logger.Warn("request DELETE", "can't query topology data from DB", topoPrefix, err_db.Error())
 	}
 
 	k8s_nodes, err1 := k8client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 
 	if err1 != nil {
-		return fmt.Errorf("INFO: list k8s host nodes info error %s", err1)
+		utils.Logger.Error("request DELETE", "k8s cluster no response", err1.Error())
+		return err1
```

Comment:
@zzengcs Same here!!

Reply:
It returns err when k8s cluster has no response on worker nodes. What's the issue here? Thank! @cj-chung